Userland/Libraries/LibWeb/CSS/Parser/Parser.cpp

```diff
@@ -4987,12 +4987,27 @@ ErrorOr<RefPtr<StyleValue>> Parser::parse_single_shadow_value(TokenStream<Compon
     auto transaction = tokens.begin_transaction();
 
     Optional<Color> color;
-    Optional<Length> offset_x;
-    Optional<Length> offset_y;
-    Optional<Length> blur_radius;
-    Optional<Length> spread_distance;
+    RefPtr<StyleValue> offset_x;
+    RefPtr<StyleValue> offset_y;
+    RefPtr<StyleValue> blur_radius;
+    RefPtr<StyleValue> spread_distance;
     Optional<ShadowPlacement> placement;
 
+    auto possibly_dynamic_length = [&](ComponentValue const& token) -> ErrorOr<RefPtr<StyleValue>> {
+        if (auto maybe_dynamic_value = TRY(parse_dynamic_value(token))) {
+            if (!maybe_dynamic_value->is_calculated())
+                return nullptr;
+            auto const& calculated_value = maybe_dynamic_value->as_calculated();
+            if (!calculated_value.resolves_to_length())
+                return nullptr;
+            return calculated_value;
+        }
+        auto maybe_spread_distance = parse_length(token);
```

Comment:
The name `maybe_spread_distance` should probably be `maybe_length`, I'm guessing that's a copy-pasta mistake. :^)

Reply:
Aha, yes you're quite right. I've now updated that :^)